spyder/plugins/editor/plugin.py

```diff
@@ -2284,6 +2273,85 @@ def new(self, fname=None, editorstack=None, text=None):
             if index is not None and not current_es.close_file(index):
                 return
 
+        # If no text is provided, create default content
+        try:
+            if text is None:
+                default_content = True
+                text, enc = encoding.read(self.TEMPLATE_PATH)
+                enc_match = re.search(r'-*- coding: ?([a-z0-9A-Z\-]*) -*-',
+                                      text)
+                if enc_match:
+                    enc = enc_match.group(1)
+
+                # Initialize template variables
+                if os.name == 'nt':
+                    # Windows
+                    import ctypes
+
+                    username = encoding.to_unicode_from_fs(
+                                os.environ.get('USERNAME', '-'))
+                    GetUserNameEx = ctypes.windll.secur32.GetUserNameExW
+                    name_display = 3
+
+                    size = ctypes.pointer(ctypes.c_ulong(0))
+                    GetUserNameEx(name_display, None, size)
+
+                    name_buffer = ctypes.create_unicode_buffer(
+                                size.contents.value)
+                    GetUserNameEx(name_display, name_buffer, size)
+                    displayname = name_buffer.value.strip()
+                else:
+                    # Linux, Mac OS X
+                    import pwd
+
+                    username = encoding.to_unicode_from_fs(
+                                   os.environ.get('USER', '-'))
+                    try:
+                        pwd_struct = pwd.getpwnam(username)
+                    except KeyError:
+                        displayname = ''
+                    else:
+                        # Use first element in Gecos field as user's full name
+                        displayname = pwd_struct.pw_gecos.split(',')[0].strip()
+
+                now = datetime.now().astimezone()
+                now.replace(microsecond=0)
+                VARS = {
+                    'date': now.ctime(),
+                    'isodate': now.isoformat(),
+                    'year': now.strftime('%Y'),
+                    'month': now.strftime('%m'),
+                    'day': now.strftime('%d'),
+                    'hour': now.strftime('%H'),
+                    'minute': now.strftime('%M'),
+                    'second': now.strftime('%S'),
+                    'tzname': now.strftime('%Z'),
+                    'monthname': now.strftime('%b'),
+                    'weekday': now.strftime('%a'),
+                    'username': username,
+                    'fullname': displayname or 'n/a',
+                    'file': fname,
+                    'filename': os.path.basename(fname),
+                    'filepath': os.path.dirname(fname),
```

Comment:
Here, I find it rather confusing that `filepath` is actually the path to the file's _parent directory_, while `file` is in fact the full path to the file itself. And may as well include the file's parent dir name, too, if we're including the path.

Also, by convention, `os.path` is imported and used as `osp` (of course, this would all be nicer using `pathlib`, but perhaps I asketh too much). Thus:

```suggestion
                    'filepath': fname,
                    'filename': osp.basename(fname),
                    'filedirpath': osp.dirname(fname),
                    'filedirname': osp.basename(osp.dirname(fname))
```

Reply:
I can see your point. I guess this is due, what you are used to: I tend to think ‘path to file’ when I here `filepath` (such as: ‘Can you please tell me the path to the file’), but I believe that in programmer speak that has a different meaning.

Since spyder is used by programmers, we probably should name the template tokens as they are used by programmers!